src/xz/header.rs

```diff
@@ -0,0 +1,111 @@
+//! XZ header.
+
+use crate::error;
+use crc::{crc32, Hasher32};
+
+/// File format magic header signature, see sect. 2.1.1.1.
+pub(crate) const XZ_MAGIC: &[u8] = &[0xFD, 0x37, 0x7A, 0x58, 0x5A, 0x00];
+
+/// Stream Header, see sect. 2.1.1.
+///
+/// This does not store the null byte in Stream Flags, which is currently unused.
+#[derive(Clone, Copy, Debug)]
+pub(crate) struct StreamHeader {
+    pub(crate) check_method: CheckMethod,
+}
+
+impl StreamHeader {
+    /// Parse a Stream Header from a byte buffer.
+    pub(crate) fn parse(input: &[u8; 12]) -> error::Result<Self> {
+        use std::hash::Hasher;
```

Comment:
I didn't check if there is a clippy lint to forbid this, but the style in `lzma-rs` is to import symbols at a module/file level.

Reply:
I'm not aware of any such clippy lints. I'll move this out.